code/drasil-theory/Theory/Drasil/ModelKinds.hs

```diff
@@ -1,42 +1,63 @@
+{-# LANGUAGE TemplateHaskell, Rank2Types, ScopedTypeVariables, PostfixOperators  #-}
+module Theory.Drasil.ModelKinds (ModelKinds(..), RealmVariant(..),
+  setMk, elimMk, lensMk, getEqModQds) where
 
-{-# LANGUAGE TemplateHaskell, Rank2Types, ScopedTypeVariables  #-}
-module Theory.Drasil.ModelKinds (ModelKinds(..), setMk, elimMk, lensMk, getEqModQds) where
-
-import Language.Drasil (QDefinition, RelationConcept, ConceptDomain(..),
-                        Definition(..), ExprRelat(..), Idea(..), 
-                        NamedIdea(..), HasUID(..))
-import Control.Lens (Lens', (^.), to, lens, set, makeLenses, Getter, Setter')
+import Control.Lens ((^.), to, lens, set, makeLenses, makeLensesFor,
+  Getter, Lens', Setter')
 import Data.Maybe (mapMaybe)
+import qualified Data.List.NonEmpty as NE
+
+import Language.Drasil (($=), sy, Expr, RelationConcept,
+  NamedIdea(..), HasUID(..), ExprRelat(..), ConceptDomain(..), Definition(..),
+  Idea(..), DefiningExpr(..), UID, Sentence, QDefinition, QuantityDict)
+
+data RealmVariant = RV {
+  _desc :: Sentence,
+  _expr :: Expr
+}
+
+makeLensesFor [("_expr", "expr")] ''RealmVariant
 
 data ModelKinds = EquationalModel QDefinition
+                | EquationalRealm QuantityDict [UID] (NE.NonEmpty RealmVariant)
```

Comment:
Are the lengths of the list and the nonempty supposed to be related?

Reply:
No, the `[UID]` is meant to be the concept domain of the variants. It's related, I think, in that it would be the union of the concept domains of the various RealmVariants. I think we discussed that we wanted it to be for the whole model instead of the individual RealmVariant, but thinking about it again from your question... it might be the case that we want both, with the top-level CD of the Realm being calculated using the CDs of the individual RealmVariants.

It also looks like the current instance of the ConceptDomain isn't aligned with this. I will fix this.